vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -796,6 +796,10 @@ public void expandRecursively(Stream<T> items, int depth) {
     public void expandRecursively(Collection<T> items, int depth) {
         getDataCommunicator()
                 .expand(getItemsWithChildrenRecursively(items, depth));
+        // Call with client roundtrip as expansion takes a while.
+        // This ensures that event is fired after exapnsion is complete.
+        getElement().executeJs("return 1;").then(result -> fireEvent(
+                new ExpandEvent<T, TreeGrid<T>>(this, false, items)));
```

Comment:
The `items` here only contains the items at the root of the recursive expansion. Sub-items that may also get expanded aren't included in the event.

Reply:
I will use null instead